sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -565,6 +565,10 @@ private void readIntBatch(int rowId, int num, WritableColumnVector column) throw
         canReadAsIntDecimal(column.dataType())) {
       defColumn.readIntegers(
           num, column, rowId, maxDefLevel, (VectorizedValuesReader) dataColumn);
+    } else if (column.dataType() == DataTypes.LongType) {
+      //  We use LongType to handle UINT32
+      defColumn.readIntegersAsUnsigned(
```

Comment:
nit: `readUnsighedIntegers`

Reply:
can we follow https://github.com/apache/spark/commit/38fbe560fd08168e90c575f7707368ddf758c3a9 and check if dictionary encoding also needs update?

Reply:
OK, checking~

Reply:
Looks irrelevant to me

Reply:
I have added the dictionary decoding code path, change the parquet data generator a bit to produce right encoded/plain data